src/Microsoft.AspNet.Mvc.Core/ActionResults/ObjectResult.cs

```diff
@@ -42,6 +42,7 @@ public ObjectResult(object value)
                 DeclaredType = DeclaredType,
                 ActionContext = context,
                 Object = Value,
+                StatusCode = StatusCode
```

Comment:
The `HttpNoContentOutputFormatter` was setting the `StatusCode` even before this change.
This change only adds the `StatusCode` that was set in the `ObjectResult` to the context so that the formatters can make a more informed decision while setting the `StatusCode`.

> This change is moving the responsibility of setting the status code from the ObjectResult to the formatter in only one case. It doesn't seem right. Perhaps just a bool saying Status Code was set? (See the flow in Object result, it sets the status code first, then calls the formatter to act).